apps/backend/src/routers/dashboard/events/teams/upload.ts

```diff
@@ -21,7 +22,8 @@ router.post(
       return;
     }
 
-    const key = `${req.event._id}/teams/${req.teamNumber}/team-info.pdf`;
+    const fileName = team.profileDocumentUrl || `${req.teamNumber}-${randomAlphanumericString(12)}`;
```

Comment:
If profile document exists - the filename will be an entire URL, and then the url will be a nested double url and break :(

Reply:
Try to extract the key from the profiledocument based on the knowledge that its the last url segment